tests/foreman/ui/test_host.py

```diff
@@ -2150,3 +2151,317 @@ def test_host_status_honors_taxonomies(
     with module_target_sat.ui_session(test_name, user=login, password=password) as session:
         statuses = session.host.host_statuses()
     assert len([status for status in statuses if int(status['count'].split(': ')[1]) != 0]) == 1
+
+
+@pytest.mark.parametrize(
+    'module_repos_collection_with_setup',
+    [
+        {
+            'distro': 'rhel8',
+            'YumRepository': {'url': settings.repos.yum_3.url},
+        }
+    ],
+    ids=['yum3'],
+    indirect=True,
+)
+@pytest.mark.parametrize('finish_via', ['rex', 'custom_rex'])
+@pytest.mark.parametrize(
+    'package_management_action',
+    [
+        'install_1_pckg',
+        'install_2_pckgs',
+        'upgrade_1_pckg',
+        'upgrade_all_pckgs',
+        'remove_1_pckg',
+        'remove_2_pckgs',
+    ],
+)
+@pytest.mark.parametrize('number_of_hosts', [1, 2], ids=['1_host', '2_hosts'])
+@pytest.mark.rhel_ver_list([settings.content_host.default_rhel_version])
+@pytest.mark.no_containers
+def test_positive_manage_packages(
+    request,
+    module_target_sat,
+    mod_content_hosts,
+    module_repos_collection_with_setup,
+    new_host_ui,
+    number_of_hosts,
+    package_management_action,
+    finish_via,
+):
+    """
+    This test is testing the new Satellite feature - Managing packages on hosts via the new All hosts UI.
+    It is highly parametrized so it can test various package management actions on various hosts.
+    Test cases are defined by the combination of the following parameters:
+    - number_of_hosts: 1 or 2
+    - package_management_action: install_1_pckg, install_2_pckgs, upgrade_1_pckg, upgrade_all_pckgs, remove_1_pckg, remove_2_pckgs
+    - finish_via: rex or custom_rex
+    All this leads to 2 * 6 * 2 = 24 test cases in total.
+    For the visual representation of the test matrix see: https://miro.com/app/board/uXjVK0pTODA=/?share_link_id=776786926115
```

Comment:
I would leave this in the PR comments, but probably not in the actual test docstring.